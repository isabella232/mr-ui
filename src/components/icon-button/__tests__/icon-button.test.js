import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './icon-button-test-cases';

describe('IconButton', () => {
  let testCase;
  let wrapper;
  let body;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
      body = shallow(
        wrapper
          .find('Tooltip')
          .props()
          .children()
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
      expect(body).toMatchSnapshot();
    });

    test('button responds to click', () => {
      body.props().onClick();
      expect(testCase.props.onClick).toHaveBeenCalledTimes(1);
    });
  });

  describe(testCases.allPropsJsxContent.description, () => {
    beforeEach(() => {
      testCase = testCases.allPropsJsxContent;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
      body = shallow(
        wrapper
          .find('Tooltip')
          .props()
          .children()
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
      expect(body).toMatchSnapshot();
    });
  });
});
