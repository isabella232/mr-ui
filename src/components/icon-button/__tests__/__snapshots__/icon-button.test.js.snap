// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton all props used, JSX content renders 1`] = `
<Tooltip
  alignment="right"
  backgroundColor="yellow"
  content={
    <span
      className="txt-bold"
    >
      tada
    </span>
  }
  disabled={false}
  display="inline-flex"
  placement="right"
  respondsToClick={true}
  testId="tooltip-test-id"
  themeTooltip="color-gray round px12 py12 shadow-darken25"
/>
`;

exports[`IconButton all props used, JSX content renders 2`] = `
<button
  className="btn px12 py12 bg-gray"
  data-test="button-test-id"
  onClick={[MockFunction]}
  type="button"
>
  <Icon
    inline={false}
    name="bike"
    themeIcon="color-gray-dark"
  />
</button>
`;

exports[`IconButton basic renders 1`] = `
<Tooltip
  alignment="center"
  backgroundColor="#fff"
  content="close"
  disabled={false}
  placement="top"
  respondsToClick={false}
  themeTooltip="round py6 px12 shadow-darken25 txt-s z1"
/>
`;

exports[`IconButton basic renders 2`] = `
<button
  className="btn btn--xs px0"
  onClick={[MockFunction]}
  type="button"
>
  <Icon
    inline={false}
    name="close"
  />
</button>
`;
